src/bounces/mod.rs

```diff
@@ -34,7 +34,7 @@ impl BounceError {
 
         BounceError {
             address: address.to_string(),
-            bounce: Some(bounce),
+            bounce: Some(bounce.clone()),
```

Comment:
Makes sense.

I'm not asking for a change here, just wondering out load: is it better to pass in a borrowed ref have it cloned locally, or to pass in a clone and take ownership of it?  I don't know if there's a standard rust pattern for approching this.

Reply:
> is it better to pass in a borrowed ref have it cloned locally, or to pass in a clone and take ownership of it?

Hmm fwiw, as a rule of thumb, cloning as late as possible makes sense to me I think. Mostly on the basis that it puts the call to `clone` closer to thing that mandates its use, but also partly because it opens the door for potential shortcut logic that doesn't need to pay for the clone (although I'm well into the territory of premature optimisation at that point).